callback_test.go

```diff
@@ -0,0 +1,83 @@
+//go:build darwin
```

Comment:
Does this test only for Darwin? What about Linux?

Reply:
Linux does not support callback. I don't have arm64 Windows.

Reply:
Windows doesn't go through this so just Darwin is fine